packaging/dev-shell.nix

```diff
@@ -1,10 +1,11 @@
 { lib, devFlake }:
 
-pkgs: stdenv:
+{ pkgs }:
 
 (pkgs.nix.override { forDevShell = true; }).overrideAttrs (attrs:
 
 let
+  stdenv = pkgs.nixDependencies.stdenv;
```

Comment:
If the shell were part of `nixComponents`, this would just be a `stdenv` parameter and it all works out.